diff --git a/dist/iac/index.js b/dist/iac/index.js
index f28c48d91c8a6df5719a697cddd88af5fb905feb..8c7bf2bb7538fdc6e09c2026b15b5352dbd62c66 100644
--- a/dist/iac/index.js
+++ b/dist/iac/index.js
@@ -273,9 +273,11 @@ function databaseRegion(database2) {
   return regions[0]?.[0];
 }
 function databaseDeploy(database2, requiredMountPath) {
+  const { multiRegionConfig, ...deployRest } = database2.deploy ?? {};
   return {
     requiredMountPath,
-    ...database2.deploy?.multiRegionConfig ? { multiRegionConfig: database2.deploy.multiRegionConfig } : {}
+    ...deployRest,
+    ...multiRegionConfig ? { multiRegionConfig } : {}
   };
 }
 function databaseToEnvironmentConfig(database2, options) {
@@ -1476,7 +1478,12 @@ function database(name, engine, options) {
     output,
     defaultMountPath: options.defaultMountPath,
     source: image(options.image),
-    ...options.region ? { deploy: { multiRegionConfig: { [options.region]: { numReplicas: 1 } } } } : {}
+    ...options.deploy || options.region ? {
+      deploy: {
+        ...options.deploy ?? {},
+        ...options.region ? { multiRegionConfig: { [options.region]: { numReplicas: 1 } } } : {}
+      }
+    } : {}
   };
   return withVariableRefs(node);
 }
diff --git a/dist/index-F4_q1IqR.d.cts b/dist/index-F4_q1IqR.d.cts
index c6676ef5526e8502378fd680319dab04fa9942d4..ac9b124927d8586071c9e700db08c4dfce01aefc 100644
--- a/dist/index-F4_q1IqR.d.cts
+++ b/dist/index-F4_q1IqR.d.cts
@@ -632,6 +632,7 @@ declare function fn<const Env extends Record<string, string | VariableConfig | V
 }): ReferencableServiceNode<RailwayProvidedVariable | Extract<keyof Env, string>>;
 interface DatabaseConfig {
     region?: string;
+    deploy?: DeployConfig;
 }
 declare function postgres(name: string, config?: DatabaseConfig): ReferencableDatabaseNode<"postgres">;
 declare function mysql(name: string, config?: DatabaseConfig): ReferencableDatabaseNode<"mysql">;
diff --git a/dist/index-F4_q1IqR.d.ts b/dist/index-F4_q1IqR.d.ts
index c6676ef5526e8502378fd680319dab04fa9942d4..ac9b124927d8586071c9e700db08c4dfce01aefc 100644
--- a/dist/index-F4_q1IqR.d.ts
+++ b/dist/index-F4_q1IqR.d.ts
@@ -632,6 +632,7 @@ declare function fn<const Env extends Record<string, string | VariableConfig | V
 }): ReferencableServiceNode<RailwayProvidedVariable | Extract<keyof Env, string>>;
 interface DatabaseConfig {
     region?: string;
+    deploy?: DeployConfig;
 }
 declare function postgres(name: string, config?: DatabaseConfig): ReferencableDatabaseNode<"postgres">;
 declare function mysql(name: string, config?: DatabaseConfig): ReferencableDatabaseNode<"mysql">;
