diff --git a/dist/iac/index.js b/dist/iac/index.js
index 1da0354260f9effb214b75270cff3c59008add16..320e75b74142e3f5ef53eef528a1803f75fd90fd 100644
--- a/dist/iac/index.js
+++ b/dist/iac/index.js
@@ -252,9 +252,11 @@ function databaseRegion(database2) {
   return regions[0]?.[0];
 }
 function databaseDeploy(database2, requiredMountPath) {
+  const { multiRegionConfig, ...deployRest } = database2.deploy ?? {};
   return {
     requiredMountPath,
-    ...database2.deploy?.multiRegionConfig ? { multiRegionConfig: database2.deploy.multiRegionConfig } : {}
+    ...deployRest,
+    ...multiRegionConfig ? { multiRegionConfig } : {}
   };
 }
 function databaseToEnvironmentConfig(database2, options) {
@@ -1398,7 +1400,12 @@ function database(name, engine, options) {
     output,
     defaultMountPath: options.defaultMountPath,
     source: image(options.image),
-    ...options.region ? { deploy: { multiRegionConfig: { [options.region]: { numReplicas: 1 } } } } : {}
+    ...options.deploy || options.region ? {
+      deploy: {
+        ...options.deploy ?? {},
+        ...options.region ? { multiRegionConfig: { [options.region]: { numReplicas: 1 } } } : {}
+      }
+    } : {}
   };
   return withVariableRefs(node);
 }
diff --git a/dist/index-Da_sOsQE.d.cts b/dist/index-Da_sOsQE.d.cts
index 39da49b6dcaa8ee913e482ed3572d9ed519b6eca..6840be0da09ad93a6b314ccd0c2b572c4303f6b7 100644
--- a/dist/index-Da_sOsQE.d.cts
+++ b/dist/index-Da_sOsQE.d.cts
@@ -624,6 +624,7 @@ declare function fn<const Env extends Record<string, string | VariableConfig | V
 }): ReferencableServiceNode<RailwayProvidedVariable | Extract<keyof Env, string>>;
 interface DatabaseConfig {
     region?: string;
+    deploy?: DeployConfig;
 }
 declare function postgres(name: string, config?: DatabaseConfig): ReferencableDatabaseNode<"postgres">;
 declare function mysql(name: string, config?: DatabaseConfig): ReferencableDatabaseNode<"mysql">;
diff --git a/dist/index-Da_sOsQE.d.ts b/dist/index-Da_sOsQE.d.ts
index 39da49b6dcaa8ee913e482ed3572d9ed519b6eca..6840be0da09ad93a6b314ccd0c2b572c4303f6b7 100644
--- a/dist/index-Da_sOsQE.d.ts
+++ b/dist/index-Da_sOsQE.d.ts
@@ -624,6 +624,7 @@ declare function fn<const Env extends Record<string, string | VariableConfig | V
 }): ReferencableServiceNode<RailwayProvidedVariable | Extract<keyof Env, string>>;
 interface DatabaseConfig {
     region?: string;
+    deploy?: DeployConfig;
 }
 declare function postgres(name: string, config?: DatabaseConfig): ReferencableDatabaseNode<"postgres">;
 declare function mysql(name: string, config?: DatabaseConfig): ReferencableDatabaseNode<"mysql">;
